### DISCUSSION TITLE
Support BTLM models from  Cerebras

### DISCUSSION THREAD

**Discussion by forrackun** [ggml-org/llama.cpp#2419] [General]
Looks like arch is bashed on Cerebras-GPT 
so should work ?
anyway i donot see any info about it if anyone gave it a try.
i always used ggml file converted by other people so have not tried.
unlike previous models bashed on Cerebras-GPT this one looks like much more capable for it's  class even challenging 7b models.

uses slimredpajama 600billion tokens cleanup from redpajama dataset

https://www.cerebras.net/blog/btlm-3b-8k-7b-performance-in-a-3-billion-parameter-model/

[Previous Cerebras-GPT Disussion ](https://github.com/ggerganov/llama.cpp/discussions/579)

**Comment by forrackun** [comment]
BTLM-3B-8K Highlights:

-    7B level model performance in a 3B model
-    State of the art 3B parameter model
-    Optimized for long sequence length inference 8K or more
-    First model trained on the [SlimPajama](https://www.cerebras.net/blog/slimpajama-a-627b-token-cleaned-and-deduplicated-version-of-redpajama), the largest fully deduplicated open dataset
-    Runs on devices with as little as 3GB of memory when quantized to 4-bit
-    Apache 2.0 license for commercial use

**Comment by forrackun** [comment]
@TheBloke  did you take a look :smile: ?

**Comment by TheBloke** [comment]
It's a new model architecture so there would need to be a GGML implementation first, at https://github.com/ggerganov/ggml - you could raise it there.

Even once a GGML implementation is added, llama.cpp is unlikely to support it for now, as currently it only supports Llama models.  Llama.cpp may add support for other model architectures in future, but not yet.

Adding a GGML implementation is not something I can do.  If a GGML implementation is released for it, I am happy to release quantisations of it.  But I can't do the implementation myself.

It will likely be easier to get GPTQ support, but again someone would have to add that, eg to AutoGPTQ.

**Comment by bornjre** [comment]
https://github.com/ggerganov/ggml/issues/427
